esy-lib/System.ml

```diff
@@ -112,4 +114,15 @@ module Environment = struct
     | Some path -> String.split_on_char sep.[0] path
     | None -> []
 
+  let homeDir () =
+    match Sys.getenv_opt "HOME", Platform.host with
+    | Some dir, _ -> dir      
+    | None, Platform.Windows -> Sys.getenv "USERPROFILE"
+    | None, _ -> raise (EnvironmentNotFound "Could not find HOME dir")
```

Comment:
Let's do `failwith` with the message instead, I don't think we can recover from this.